### PR TITLE
Issue 5670 tool strip buttons rendering tests

### LINKED ISSUE #5670: Write rendering tests for ToolStrip Button background rendering fix
Generally we should write [rendering tests](https://github.com/dotnet/winforms/blob/main/docs/testing.md#rendering-tests), if possible. There are a few example in the codebase, e.g. https://github.com/dotnet/winforms/pull/4739.

_Originally posted by @RussKie in https://github.com/dotnet/winforms/issues/5614#issuecomment-909859179_

### PR DESCRIPTION
**Pull request by ArtemTatarinov**
Fixes #5670

## Proposed changes

- Added rendering tests for ToolStrip buttons

## Customer Impact

- No

## Regression? 

- No

## Risk

- Minimal

## Test environment(s) 
Microsoft Windows [Version 10.0.19043.1165]
.NET SDK 6.0.100-rc.1.21416.15

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/ToolStripSystemRenderer.cs [hunk 1]
```diff
@@ -42,7 +42,8 @@ internal ToolStripRenderer HighContrastRenderer
             {
                 if (toolStripHighContrastRenderer is null)
                 {
-                    toolStripHighContrastRenderer = new ToolStripHighContrastRenderer(/*renderLikeSystem*/true);
+                    // If system in high contrast mode 'false' flag should be passed to render filled selected button background. This is in consistence with ToolStripProfessionalRenderer.
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/ToolStripSystemRenderer.cs:45]:
```suggestion
                    // If system in high contrast mode 'false' flag should be passed to render filled selected button background.
                    // This is in consistence with ToolStripProfessionalRenderer.
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.Rendering.cs [hunk 2]
```diff
@@ -58,5 +58,213 @@ public void ToolStrip_RendersBackgroundCorrectly()
...
+
+            // Force the handle creation
+            Assert.NotEqual(IntPtr.Zero, form.Handle);
+            Assert.NotEqual(IntPtr.Zero, toolStrip.Handle);
+
+            using EmfScope emf = new();
+            DeviceContextState state = new(emf);
+
+            Rectangle bounds = toolStrip.Bounds;
+            PaintEventArgs e = new(emf, bounds);
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.Rendering.cs:80]:
```suggestion
            using PaintEventArgs e = new(emf, bounds);
```
ditto below

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.Rendering.cs [hunk 3]
```diff
@@ -58,5 +58,213 @@ public void ToolStrip_RendersBackgroundCorrectly()
...
+            toolStrip.TestAccessor().Dynamic.OnPaintBackground(e);
+
+            Rectangle bitBltBounds = new(bounds.X, bounds.Y, bounds.Width - 1, bounds.Height - 1);
+
+            var details = emf.RecordsToString();
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.Rendering.cs:85]: This is not necessary in automated tests.
Ditto below

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.Rendering.cs [hunk 4]
```diff
@@ -58,5 +58,213 @@ public void ToolStrip_RendersBackgroundCorrectly()
...
+               Validate.Polygon16(
+                    bounds: null,
+                    points: null,
+                    State.Brush(SystemColors.Highlight, Gdi32.BS.SOLID)),
+               Validate.Repeat(Validate.SkipType(Gdi32.EMR.POLYGON16), 3)
+              );
+        }
+    }
+
+    class ToolStripSystemHighContrastRenderer : ToolStripSystemRenderer
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.Rendering.cs:266]:
```suggestion
    private class ToolStripSystemHighContrastRenderer : ToolStripSystemRenderer
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.Rendering.cs [hunk 5]
```diff
@@ -58,5 +58,213 @@ public void ToolStrip_RendersBackgroundCorrectly()
...
+                state,
+                Validate.BitBltValidator(
+                    bitBltBounds,
+                    State.BrushColor(Color.Blue)),
+                Validate.SkipType(Gdi32.EMR.BITBLT)
+               );
+        }
+
+        [WinFormsFact]
+        public void ToolStrip_Button_RendersTextCorrectly()
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripTests.Rendering.cs:97]: Tests that test `ToolStripButton` painting (as opposed to `ToolStrip` painting) should be located in `ToolStripButtonTests.Rendering` as these tests test the button's implementations and not the toolstip.

**Review by RussKie — CHANGES_REQUESTED** [review]
I'm struggling to reconcile the "fixes 5670" + the subject against the actual implementation. That is, changes under .\tests folder are expected, but changes under .\src folder are not.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripButtonTests.Rendering.cs [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+            using ToolStrip toolStrip = new();
+            using ToolStripButton toolStripButton = new()
+            {
+                Text = "Hello"
+            };
+            toolStrip.Items.Add(toolStripButton);
+            form.Controls.Add(toolStrip);
+
+            using EmfScope emf = new();
+
```
Comment (dreddy-work) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/ToolStripButtonTests.Rendering.cs:27]: nit: extra lines not consistent with other tests.

**Review by dreddy-work — COMMENTED** [review]
LGTM.

**Review by RussKie — APPROVED** [review]
👍

**Pull request merged**